testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppWindow.kt

```diff
@@ -0,0 +1,183 @@
+
+        if (!isAccepted()) {
+            launchScreen()
+        } else if (!doesDataExist()) {
+            storeAcceptance(DEFAULT_ACCEPTANCE) // should be false only after accept.
+            launchScreen()
+        } else {
+            preloadMiniAppLaunchListener.onPreloadMiniAppResponse(true)
+        }
+    }
+
+    private fun launchScreen() {
+        initDefaultWindow()
+        preloadMiniAppAlertDialog.show()
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun initDefaultWindow() {
+        // set ui components
+        val layoutInflater = LayoutInflater.from(context)
+        preloadMiniAppLayout = layoutInflater.inflate(R.layout.window_preload_miniapp, null)
+        preloadMiniAppAlertDialog = AlertDialog.Builder(context, R.style.AppTheme_DefaultWindow).create()
+        preloadMiniAppAlertDialog.setView(preloadMiniAppLayout)
+
+        // set data to ui
+        val nameView = preloadMiniAppLayout.findViewById<TextView>(R.id.preloadMiniAppName)
+        val versionView = preloadMiniAppLayout.findViewById<TextView>(R.id.preloadMiniAppVersion)
+        if (miniAppInfo != null) {
+            setIcon(
+                context,
+                Uri.parse(miniAppInfo?.icon),
+                preloadMiniAppLayout.findViewById(R.id.preloadAppIcon)
+            )
+
+            nameView.text = miniAppInfo?.displayName.toString()
+            versionView.text = "Version: " + miniAppInfo?.version?.versionTag.toString()
+        } else {
+            nameView.text = "No info found for this miniapp!"
+        }
+
+        // set manifest/metadata to UI: permissions
+        val permissionAdapter = PreloadMiniAppPermissionAdapter()
+        preloadMiniAppLayout.findViewById<RecyclerView>(R.id.listPreloadPermission).layoutManager =
+            LinearLayoutManager(context)
+        preloadMiniAppLayout.findViewById<RecyclerView>(R.id.listPreloadPermission).adapter =
+            permissionAdapter
+        preloadMiniAppLayout.findViewById<RecyclerView>(R.id.listPreloadPermission)
+            .addItemDecoration(
+                DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
+            )
+
+        viewModel =
+            ViewModelProvider.NewInstanceFactory().create(PreloadMiniAppViewModel::class.java)
+                .apply {
+                    // observe version id when it's empty
+                    if (versionId.isEmpty()) {
+                        miniAppVersionId.observe(lifecycleOwner, Observer { versionId = it })
+                        versionIdErrorData.observe(lifecycleOwner, Observer {
+                            Toast.makeText(context, it, Toast.LENGTH_LONG).show()
+                        })
+                    }
+
+                    miniAppManifestMetadata.observe(lifecycleOwner, Observer {
+                        preloadMiniAppLayout.findViewById<TextView>(R.id.preloadMiniAppMetaData).text =
+                            "Metadata value: " + it
```

Comment:
Should this be a string resource instead? Also, I think text like `Custom MetaData` might be better to match the naming in `manifest.json`.